### PR TITLE
(JavaFXSceneBuilder): Init at version 2

### PR DESCRIPTION
**Pull request by ysndr**
###### Motivation for this change
Add JavaFXSceneBuilder to be used standalone or inside intellij or other IDEs

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/javafx-scene-builder/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  name = "${pname}-${version}";
+
+  src = requireFile {
+    name = "javafx_scenebuilder-${version}-linux-x64.tar.gz";
+      sha256 = "0b66wxrg9n6i0ysvl4n07kj1d5zqk32vywxrvdn3spykn8pxixj4";
+    message = ''
+      SceneBuilder cannot be downloaded without license acception.
+      1. Download a copy from https://www.oracle.com/technetwork/java/javase/downloads/javafxscenebuilder-1x-archive-2199384.html
+      2. In a shell, execute:
+         nix-prefetch-url file:///path/to/Xcode_7.2.dmg
```
Comment (matthewbauer) [inline pkgs/applications/editors/javafx-scene-builder/default.nix]: Make sure to update this text for javascenebuilder

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by ysndr** [comment]
Have not been in need of this since then and have totally forgotten about it. I'll look into this. I think the only issue has been the description. Now, maybe a fresh rebase might be needed, but apart from that this should be fine.

**Comment by ysndr** [comment]
Actually `autoPatchelf` is not able to find `libavcodec.so.53` and `libavformat.so.53`. Any idea how to include those? Neither importing `ffmpeg` nor `libav` respectively would fix this. :thinking:

File: pkgs/applications/editors/javafx-scene-builder/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+libxslt,
+libxml2,
+libXtst,
+libXxf86vm
+}:
+
+stdenv.mkDerivation rec {
+  pname = "JavaSceneBuilder";
+  version = "2_0";
+  name = "${pname}-${version}";
```
Comment (ryantm) [inline pkgs/applications/editors/javafx-scene-builder/default.nix]:
```suggestion
```

File: pkgs/applications/editors/javafx-scene-builder/default.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  src = requireFile {
+    name = "javafx_scenebuilder-${version}-linux-x64.tar.gz";
+      sha256 = "0b66wxrg9n6i0ysvl4n07kj1d5zqk32vywxrvdn3spykn8pxixj4";
+    message = ''
+      SceneBuilder cannot be downloaded without license acception.
```
Comment (ryantm) [inline pkgs/applications/editors/javafx-scene-builder/default.nix]:
```suggestion
      SceneBuilder cannot be downloaded without accepting a license.
```

**Review by ryantm — CHANGES_REQUESTED** [review]
Currently this package is causing an evaluation error with ofborg. So that needs to be resolved.

The library issue looks similar to what is going on here: https://github.com/NixOS/nixpkgs/blob/9842c4b1076a2f54de5e9a81fd91b2b91b1d051e/pkgs/applications/audio/spotify/default.nix#L126

**Comment by ysndr** [comment]
I see that oracle now requires one to register an account on order to download the scene builder. I also wonder if this program is even used that broadly? If not I think this PR could be closed, and I might look into packaging [gluon](https://gluonhq.com//products/scene-builder) instead if there is interest.

**Comment by ShamrockLee** [comment]
As far as I know, SceneBuilder is now distributed under the 3-clause BSD License on GitHub, and the installers can also be downloaded from the official website directly (without registrations or something similar).
GitHub page: [ gluonhq/scenebuilder ](https://github.com/gluonhq/scenebuilder)
Home page: https://gluonhq.com/products/scene-builder/
It would be awsome to have it as a Nix package.

**Comment by SuperSandro2000** [comment]
Replaced by #110398

**Pull request closed without merging**